src/app/zap-templates/zcl/data-model/chip/switch-cluster.xml

```diff
@@ -71,63 +74,63 @@ Interactions with the switch device are exposed as attributes (for the latching
-    </attribute>
-    <attribute side="server" code="0x0001" name="CurrentPosition" define="CURRENT_POSITION" type="int8u" reportable="true">
-      <mandatoryConform/>
-    </attribute>
-    <attribute side="server" code="0x0002" name="MultiPressMax" define="MULTI_PRESS_MAX" type="int8u" optional="true" default="2" min="2">
+    <attribute side="server" code="0x0000" name="NumberOfPositions" define="NUMBER_OF_POSITIONS" type="int8u" min="2"/>
+    <attribute side="server" code="0x0001" name="CurrentPosition" define="CURRENT_POSITION" type="int8u" max="254"/>
+    <attribute side="server" code="0x0002" name="MultiPressMax" define="MULTI_PRESS_MAX" type="int8u" min="2">
       <mandatoryConform>
         <feature name="MSM"/>
       </mandatoryConform>
     </attribute>
-    <event side="server" code="0x00" priority="info" name="SwitchLatched" optional="true">
-      <description>SwitchLatched</description>
-      <field id="0" name="NewPosition" type="int8u" />
+    <event side="server" code="0x00" priority="info" name="SwitchLatched">
+      <description>This event SHALL be generated, when the latching switch is moved to a new position.</description>
+      <field id="0" name="NewPosition" type="int8u" max="254"/>
       <mandatoryConform>
         <feature name="LS"/>
       </mandatoryConform>
     </event>
-    <event side="server" code="0x01" priority="info" name="InitialPress" optional="true">
-      <description>InitialPress</description>
-      <field id="0" name="NewPosition" type="int8u" />
+
+    <event side="server" code="0x01" priority="info" name="InitialPress">
+      <description>This event SHALL be generated, when the momentary switch starts to be pressed (after debouncing).</description>
+      <field id="0" name="NewPosition" type="int8u" max="254"/>
       <mandatoryConform>
         <feature name="MS"/>
       </mandatoryConform>
     </event>
-    <event side="server" code="0x02" priority="info" name="LongPress" optional="true">
-      <description>LongPress</description>
-      <field id="0" name="NewPosition" type="int8u" />
+
+    <event side="server" code="0x02" priority="info" name="LongPress">
+      <description>This event SHALL be generated when the momentary switch has been pressed for a &quot;long&quot; time.</description>
+      <field id="0" name="NewPosition" type="int8u" max="254"/>
       <mandatoryConform>
         <feature name="MSL"/>
       </mandatoryConform>
     </event>
-    <event side="server" code="0x03" priority="info" name="ShortRelease" optional="true">
-      <description>ShortRelease</description>
-      <field id="0" name="PreviousPosition" type="int8u" />
+
+    <event side="server" code="0x03" priority="info" name="ShortRelease">
+      <description>If the server has the Action Switch (AS) feature flag set, this event SHALL NOT be generated at all, since setting the Action Switch feature flag forbids the Momentary Switch ShortRelease (MSR) feature flag from being set.</description>
```

Comment:
This description is a secondary aspect, and not the primary reason the event exists. This looks like a scraping issue.